### ISSUE TITLE
Parameter parsing/passing: The CLI's -File parameter doesn't support negative numbers as positional arguments

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#17519]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Note: This is related to various other, similar bugs: https://github.com/PowerShell/PowerShell/issues?q=is%3Aissue+is%3Aopen+%22parameter+parsing%2Fpassing%22

When using `pwsh -File` trying to pass a _positional_ argument that is a _negative number_ fails, because the `-` is apparently invariably interpreted as starting a parameter _name_:

```powershell
# Create sample script t.ps1
@'
[CmdletBinding()] param([int] $p) "-p value: $p"
'@ > t.ps1

pwsh -NoProfile -File t.ps1 -1     # !! BREAKS: *positional* argument
pwsh -NoProfile -File t.ps1 -p -1  # OK, as *named* argument.
./t.ps1 -1                         # OK, positional argument with *direct invocation*.
```

Note that quoting `-1` does _not_ help.

### Expected behavior

```console
-p value: -1
-p value: -1
-p value: -1
```

### Actual behavior

```console
t.ps1: A parameter cannot be found that matches parameter name '1'.
-p value: -1
-p value: -1
```

### Environment data

```powershell
PowerShell Core v7.3.0-preview.4
```

**Comment by scriptingstudio** [comment]
Windows 11 + PS7.2.4 - no errors

does this work as a workaround?
`pwsh -NoProfile -File t.ps1 (-1)`

**Comment by mklement0** [comment]
That's surprising, because the error is related to PowerShell's parameter parsing, which isn't OS-specific. I don't have Windows 11, but I see the error in  WinPS, 7.2.4, and  v7.3.0-preview.4, both on Windows 10 and macOS 12.4.

Your attempted workaround is syntactically only an option from _inside_ PowerShell, where there's rarely a need to call the CLI (if you needed to, you would use `pwsh -NoProfile { ./t.ps1 -1 }`, which avoids the problem); even there, however, it isn't effective, because the `(1)` gets translated to a _string_ with value `1`, given that a call to an _external program_ is being made.

**Comment by scriptingstudio** [comment]
Tested it on another Windows 11 box - no errors.
tested on Windows Server 2016 - no errors.

**Comment by mklement0** [comment]
@scriptingstudio, just to narrow things down and rule out that you're not accidentally testing something different from the repro in the OP:

In a shell started with  `powershell.exe -NoProfile`, i.e., _Windows PowerShell_, I see the error too - do you not?
(The specific WinPS version is v5.1.19041.1682, though the specific version is unlikely to matter, given that Windows PowerShell is unlikely to have received a fix.)

I see the error on Windows 10, macOS 12.4, and Ubuntu 18.04 - again suggesting that the OS is incidental.

**Comment by Pablo1q** [comment]
Hello , i   am Pablo and working was good for regular and fix. I want the bot again!, Hope we do It again. Good Luck ALL the best

**Comment by scriptingstudio** [comment]
@mklement0  I got it, could repro. I think it is `pwsh`/`powershell` parameter binder problem (or something like that). It considers an argument with `-` as a parameter

**Comment by SteveL-MSFT** [comment]
Unfortunately, it appears to be `by design` based on this code https://github.com/PowerShell/PowerShell/blob/master/src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs#L1352-L1383.  It's doing a simple parameter binding which is a subset of the actual PS parameter binder.

I'd be worried what scenarios would break if we try to change this.

**Comment by mklement0** [comment]
@SteveL-MSFT, calling an _implementation shortcut_ that is inconsistent with how parameter names are normally parsed _by design_  strikes me as overly generous (unavoidable differences between `-File`-based and in-session / `-Command` notwithstanding).

As for what could break if something like `-1` were no longer considered a parameter _name_ but a _positional argument_ (which is how it's interpreted in-session):

* Note that _in-session_ / with `-Command`, a script's parameter that looks like a negative number, such as `-1` (`$1`) cannot be bound by a _named_ argument.

   * That is, such a parameter can only be bound _positionally_, except via splatting - an inconsistency in its own right; see #17767

```powershell
# !! BREAKS: "-1" is interpreted as a positional argument:
# !!   -> A positional parameter cannot be found that accepts argument 'foo'
 & { [CmdletBinding()] param([string]$1) "[$1]" } -1 foo
```

* However, because of the bug at hand, this _does_ work - but arguably shouldn't - when called via the CLI's `-File` parameter.

```powershell
'[CmdletBinding()] param([string]$1) "[$1]"' > t.ps1

# -> 'really?'
pwsh -nop -file t.ps1 -1 really?
```

In other words: the following code could break:

* calls to scripts declared with ill-advised parameter names such as `-1` (parameter variable `$1`)
* _if_ they are called with a _named_ argument binding such a parameter _but only ever via the CLI's `-File` parameter_, given that this otherwise doesn't work.

Perhaps needless to say, the existence of such scripts / calls seems very unlikely.

Conversely, not being able to pass a negative number via `-File` is an unnecessary restriction, especially given that the functionality that implicitly prevents it shouldn't exist to begin with: if `-1 foo` doesn't work in a PowerShell session, neither should it via `-File`
